src/finalize.cpp

```diff
@@ -198,4 +197,11 @@ int openmc_hard_reset()
   // Reset the random number generator state
   openmc::openmc_set_seed(DEFAULT_SEED);
   return 0;
+
+void openmc_initialize_geometry()
+{
+  free_memory_geometry();
+  free_memory_surfaces();
+  read_geometry_xml();
+  finalize_geometry();
```

Comment:
While this may work for some situations, in others it may cause problems. Namely, looking at the `read_input_xml` function, after loading the geometry, `finalize_cross_sections` loads cross sections based on the temperatures that are in the model. Also, `prepare_distribcells` sets up offsets for use of distributed cells. If these are not called again after reinitializing the geometry, things might get messed up.

Would exposing the call the `read_input_xml` work for your use case? Is there some information you want to persist by only reinitializing geometry?

Reply:
Hi Paul, 
Thanks for you comment. 
I understand it might seem counter-intuitive, but I needed a function to reinitialize the geometry only, based on some changes that happen to the `geometry.xml` file in a depletion run. 
Basically, one would initialize with `read_input_xml` at the first iteration and later, if wanted, reinitialize a new `geometry.xml` with this new function.
My particular need is a depletion case with a geometry feature that I allow to change at every time-step (for example to control reactivity) and that I need to re-initialize iteratively to the run (in memory), though without reinitialize all the rest. 
I hope I clarified a bit more.